lesson_26/api/java/api_app/src/main/java/com/codedifferently/lesson26/web/MediaItemsController.java

```diff
@@ -29,4 +34,31 @@ public GetMediaItemsResponse getItems() {
     var response = GetMediaItemsResponse.builder().items(responseItems).build();
     return response;
   }
+
+  @PostMapping("/items")
+  public MediaItemResponse addItemById(@RequestBody CreateMediaItemRequest createRequest) {
+    if (createRequest.getItem().getId() == null) {
+      createRequest.getItem().setId(UUID.randomUUID());
+    }
+    
+      library.add(createRequest.getItem());
+    
+    return  MediaItemResponse.from(createRequest.getItem());
+  }
+
+  @GetMapping("/items/:{id}")
```

Comment:
The path should not include the semicolon in this case (in some frameworks, they use semicolon so make sure to know what to do in the right context).